PR: https://github.com/odoo/odoo/pull/121222

From: a9b5f5c3c1f6665733d22c7f93ddecf0141ecc84
From: Jurgen (jugj)
Date: 2023-05-16 11:37:05

Structural Changes: 6
Total Changes: 39

[IMP] hr_recruitment : Improve the UX of recruitment app

task - 3175879

closes odoo/odoo#121222

X-original-commit: 5625c013f86f3ae2617614ccde3b789bbd5ab169
Related: odoo/enterprise#40965
Signed-off-by: Kevin Baptiste <kba@odoo.com>
Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_recruitment/models/calendar.py
+++ b/addons/hr_recruitment/models/calendar.py
@@ -14,7 +14,9 @@ class CalendarEvent(models.Model):
             self = self.with_context(
                 default_res_model='hr.applicant', #res_model seems to be lost without this
                 default_res_model_id=self.env.ref('hr_recruitment.model_hr_applicant').id,
-                default_res_id=self.env.context['default_applicant_id']
+                default_res_id=self.env.context['default_applicant_id'],
+                default_partner_ids=self.env.context['default_partner_ids'],
+                default_name=self.env.context['default_name']
             )
 
         defaults = super(CalendarEvent, self).default_get(fields)

--- a/addons/hr_recruitment/models/hr_applicant.py
+++ b/addons/hr_recruitment/models/hr_applicant.py
@@ -30,7 +30,7 @@ class Applicant(models.Model):
     name = fields.Char("Subject / Application", required=True, help="Email subject for applications sent via email", index='trigram')
     active = fields.Boolean("Active", default=True, help="If the active field is set to false, it will allow you to hide the case without removing it.")
     description = fields.Html("Description")
-    email_from = fields.Char("Email", size=128, help="Applicant email", compute='_compute_partner_phone_email',
+    email_from = fields.Char("Email", size=128, compute='_compute_partner_phone_email',
         inverse='_inverse_partner_email', store=True, index='trigram')
     probability = fields.Float("Probability")
     partner_id = fields.Many2one('res.partner', "Contact", copy=False)
@@ -50,7 +50,7 @@ class Applicant(models.Model):
     date_closed = fields.Datetime("Hire Date", compute='_compute_date_closed', store=True, readonly=False, tracking=True)
     date_open = fields.Datetime("Assigned", readonly=True)
     date_last_stage_update = fields.Datetime("Last Stage Update", index=True, default=fields.Datetime.now)
-    priority = fields.Selection(AVAILABLE_PRIORITIES, "Appreciation", default='0')
+    priority = fields.Selection(AVAILABLE_PRIORITIES, "Evaluation", default='0')
     job_id = fields.Many2one('hr.job', "Applied Job", domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]", tracking=True, index=True)
     salary_proposed_extra = fields.Char("Proposed Salary Extra", help="Salary Proposed by the Organisation, extra advantages", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
     salary_expected_extra = fields.Char("Expected Salary Extra", help="Salary Expected by Applicant, extra advantages", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
@@ -343,11 +343,11 @@ class Applicant(models.Model):
             alias_id = False
 
         nocontent_body = Markup("""
-<p class="o_view_nocontent_empty_folder">%(help_title)s</p>
+<p class="o_view_nocontent_smiling_face">%(help_title)s</p>
 <p>%(para_1)s<br/>%(para_2)s</p>""") % {
-            'help_title': _('No application yet'),
-            'para_1': _('Let people apply by email to save time.'),
-            'para_2': _('Attachments, like resumes, get indexed automatically.'),
+            'help_title': _("No application found. Let's create one !"),
+            'para_1': _('People can also apply by email to save time.'),
+            'para_2': _("You can search into attachment's content, like resumes, with the searchbar."),
         }
 
         if alias_id and alias_id.alias_domain and alias_id.alias_name:
@@ -379,6 +379,19 @@ class Applicant(models.Model):
             @return: Dictionary value for created Meeting view
         """
         self.ensure_one()
+
+        if not self.partner_id:
+            if not self.partner_name:
+                raise UserError(_('You must define a Contact Name for this applicant.'))
+            self.partner_id = self.env['res.partner'].create({
+                'is_company': False,
+                'type': 'private',
+                'name': self.partner_name,
+                'email': self.email_from,
+                'phone': self.partner_phone,
+                'mobile': self.partner_mobile
+            })
+
         partners = self.partner_id | self.user_id.partner_id | self.department_id.manager_id.user_id.partner_id
 
         category = self.env.ref('hr_recruitment.categ_meet_interview')
@@ -424,7 +437,8 @@ class Applicant(models.Model):
             'view_mode': 'tree,kanban,form,pivot,graph,calendar,activity',
             'domain': [('id', 'in', other_applicants.ids)],
             'context': {
-                'active_test': False
+                'active_test': False,
+                'search_default_stage': 1,
             },
         }
 

--- a/addons/hr_recruitment/models/hr_job.py
+++ b/addons/hr_recruitment/models/hr_job.py
@@ -38,7 +38,7 @@ class Job(models.Model):
     hr_responsible_id = fields.Many2one(
         'res.users', "HR Responsible", tracking=True,
         help="Person responsible of validating the employee's contracts.")
-    document_ids = fields.One2many('ir.attachment', compute='_compute_document_ids', string="Documents")
+    document_ids = fields.One2many('ir.attachment', compute='_compute_document_ids', string="Documents", readonly=True)
     documents_count = fields.Integer(compute='_compute_document_ids', string="Document Count")
     alias_id = fields.Many2one(
         'mail.alias', "Alias", ondelete="restrict", required=True,
@@ -260,10 +260,9 @@ class Job(models.Model):
                 'default_res_id': self.ids[0],
                 'show_partner_name': 1,
             },
-            'view_mode': 'tree,form',
+            'view_mode': 'tree',
             'views': [
-                (self.env.ref('hr_recruitment.ir_attachment_hr_recruitment_list_view').id, 'tree'),
-                (False, 'form'),
+                (self.env.ref('hr_recruitment.ir_attachment_hr_recruitment_list_view').id, 'tree')
             ],
             'search_view_id': self.env.ref('hr_recruitment.ir_attachment_view_search_inherit_hr_recruitment').ids,
             'domain': ['|',
